### PR TITLE
use travis-CI to test our images

### LINKED ISSUE #58: Add Travis CI support
@maxGimeno 

Problems with Docker Hub automatic build
-------
For the moment, for https://github.com/CGAL/cgal-testsuite-dockerfiles we use the Docker service integration, but it is pretty bad.

- All images are rebuild sequential, but in an order that we do not control. We have images that depend on other images, such as the image build with `ArchLinux-CXX14/Dockerfile` that depends on the image build with `ArchLinux/Dockerfile`. 

- Docker service integration does not test pull requests.

- Docker service integration does not give status feedback to Github.
______________
Solution: Travis CI
-----
That is why I would like that we try Travis CI for `CGAL/cgal-testsuite-dockerfiles`. Using a [build matrix], we could build several images in parallel. However, a chain of dependent images will have to be build sequentially in the same job.
[build matrix]: https://docs.travis-ci.com/user/customizing-the-build/#Build-Matrix

Additional:
- Using [encrypted secret files] to store authentification tokens, we could also deploy the images to the Docker Registry.
- Using Travis CI, we could even add tests to our building of images. But I do not know what tests we could do.

[encrypted secret files]: https://docs.travis-ci.com/user/encryption-keys/
(Cc: @afabri That is the project I talked about.)

### PR DESCRIPTION
**Pull request by maxGimeno**
This PR fixes #58 .
It uses a .travis.yml file to perform a battery of tests on the docker images.

### CODE REVIEW

File: test.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash 
+
+if [ "$1" == --Fedora ]
```
Comment (lrineau) [inline test.sh]: > `if [ "$1" == --Fedora ]`

It should be:
```shell
if [ "$1" == Fedora ]
```

**Comment by lrineau** [comment]
You should rebase your branch onto `master`. In the builds of CentOS 6 and -32, I have passed `-d0` to Boost b2 tool: that reduces the size of logs.

File: .travis.yml [hunk 2]
```diff
@@ -19,3 +19,6 @@ script :
 - chmod +x ./test.sh 
 - travis_wait 120 ./test.sh $PLATFORM
 
+after_success:
+- chmod +x ./push.sh
+- - if [ "$TRAVIS_BRANCH" == "master" ]; then  docker login -u $USER -p $PASS; ./push.sh $PLATFORM fi
```
Comment (lrineau) [inline .travis.yml]: The variable `USER` and `PASS` must be set using the encryption features of Travis.

Comment (maxGimeno) [inline .travis.yml]: Indeed, I forgot that obvious part !

Comment (lrineau) [inline .travis.yml]: To test it, do the `docker login` before the test of `$TRAVIS_BRANCH`.

**Comment by lrineau** [comment]
Maxime: I cannot remove the branch `test_images_with_travisCI-GF` in your repo. You can now remove it.

**Pull request merged**